src/config/readConfigFile.ts

```diff
@@ -5,13 +5,36 @@
 import fs from "fs";
 import { buildReadConfigFileCallback } from "./buildReadConfigFileCallback";
 import { OpenResult } from "./openConfig";
+import path from "path";
+import { CheckedConfig } from "./execute";
 
 // eslint-disable-next-line functional/no-return-void, @typescript-eslint/prefer-readonly-parameter-types
 export type Accept = (value: OpenResult | PromiseLike<OpenResult>) => void;
 
 export const configFile = ".folder-structure-lint.json";
-export const message = `Could not open the config file "${configFile}"`;
+export const jsConfigFile = ".folder-structure-lint.js";
+export const message = `Could not open the config file "${configFile}" or "${jsConfigFile}"`;
 
+type MaybeCheckedConfig = Partial<CheckedConfig>;
+
+// eslint-disable-next-line max-lines-per-function
 export function readConfigFile(accept: Accept) {
-  fs.readFile(configFile, "utf8", buildReadConfigFileCallback(accept));
+  if (fs.existsSync(jsConfigFile)) {
+    const absolutePath = path.resolve(jsConfigFile);
+
+    // eslint-disable-next-line functional/no-try-statement
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const config = require(absolutePath) as MaybeCheckedConfig;
```

Comment:
Testing this is also non-trivial because we can't spyOn the `require` statement. Any ideas? I can mock the implementation of the `existsSync` function so this branch gets tested as well, but I am at a loss with regards to confirming the `require` statement is called with the correct argument. 

I could test the return of the `readConfigFile` as an alternative...

Reply:
We can merge this before deciding on a test strategy too. I will look at coverage before publishing though, and hand test then for sure.